### PR TITLE
Add uploaderName as DBS3Upload config parameter

### LINKED ISSUE #11494: Make DBS objects `created_by` attribute configurable DBSUpload
**Impact of the new feature**
T0, WMAgent, DBS

**Is your feature request related to a problem? Please describe.**
Currently, there is no direct way to distinguish files created by T0, from files created by central production. Similarly, it is not possible to directly filter Replay generated files from production files. Thius is because the `created_by` field in DBS is always set to `WMAgent`, as can be seen here: [/WMComponent/DBS3Buffer/DBSBufferBlock.py#L70](https://github.com/dmwm/WMCore/blob/fea1de4119a9091233f3e6414f33f069971b5a74/src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py#L70)

**Describe the solution you'd like**
Created a new DBSUpload parameter in the WMAgent configuration called `UploaderName`, that defaults to `WMAgent`, but can be configured to `T0Prod` or `T0Replay`. This attribute would be used to fill the `created_by` and `last_modified_by` fields on DBSUpload.

**Describe alternatives you've considered**
Other existing fields can be use to make to distinguish T0 and Central production files, like origin_site_name, as T0 uses T0_CH_CERN_Disk. Nonetheless, this correlation is circunstantial and cannot be guaranteed, specially when it comes to Replay data. Other combinations of filters can be used in specific situations, like keywords in dataset names or lfn, but we would like to have a direct disctinction at file level.

**Additional context**

### PR DESCRIPTION
**Pull request by germanfgv**
Fixes #11494 

#### Status
ready

#### Description
Add uploaderName as DBS3Upload config parameter. This is used in DBSBufferBlock to fill the `create_by` and `last_modified_by` attributes in DBS files and datasets

#### Is it backward compatible (if not, which system it affects?)
YES

### CODE REVIEW

**Comment by germanfgv** [comment]
Here you can see the files uploaded by the replay testing these changes.
https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=dataset%3D%2F*%2F*v23194233%2F*

**Review by amaltaro — COMMENTED** [review]
@germanfgv German, please let me know when you manage to follow up on what we discussed in the meeting yesterday.

File: src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py [hunk 1]
```diff
@@ -62,12 +62,13 @@ def __init__(self, name, location, datasetpath):
         self.location     = location
         self.datasetpath  = datasetpath
         self.workflows    = set()
+        self.uploader      = uploader
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py:65]: Even though the previous lines have this indented assignment, could you please surround the assignment by a single spacing, e.g.: `        self.uploader = uploader`

File: src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py [hunk 2]
```diff
@@ -189,6 +189,11 @@ def __init__(self, config):
         self.physicsGroup = getattr(self.config.DBS3Upload, "physicsGroup", "NoGroup")
         self.datasetType = getattr(self.config.DBS3Upload, "datasetType", "PRODUCTION")
         self.primaryDatasetType = getattr(self.config.DBS3Upload, "primaryDatasetType", "mc")
+        self.uploaderName = getattr(self.config.DBS3Upload, "uploaderName", "WMAgent")
+        if self.uploaderName not in ("WMAgent", "T0Prod", "T0Replay"):
+            message="Invalid value for attribute uploaderName: " + self.uploaderName
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py:194]: Maybe refactor these lines with:
```
raise DBSUploadException(f"Invalid value for attribute uploaderName: {self.uploaderName}")
```

**Review by amaltaro — COMMENTED** [review]
@germanfgv this looks pretty good. I have made 2 minor suggestions in the code though that would improve it.
Once you have changes in place, an you please squash these commits accordingly (src changes in one commit, test changes in another)? Thanks!

**Review by amaltaro — APPROVED** [review]
Thanks German! It looks good to me!

**Pull request merged**